### ISSUE TITLE
Unable to delete BigQuery Database

### ISSUE THREAD

**Issue by tobbbles** [metabase/metabase#5696]
When trying to remove a BQ database via the admin I'm greeted with the following error:

```
Cannot delete or update a parent row: a foreign key constraint fails (`metabase`.`metabase_field`, CONSTRAINT `fk_field_ref_table_id` FOREIGN KEY (`table_id`) REFERENCES `metabase_table` (`id`))
```

Full error log:
```
08-07 11:56:25 ERROR metabase.middleware :: DELETE /api/database/6 500 (48 s) (2425 DB calls)
{:message
 "Cannot delete or update a parent row: a foreign key constraint fails (`metabase`.`metabase_field`, CONSTRAINT `fk_field_ref_table_id` FOREIGN KEY (`table_id`) REFERENCES `metabase_table` (`id`))",
 :stacktrace
 ["models.database$pre_delete.invokeStatic(database.clj:52)"
  "models.database$pre_delete.invoke(database.clj:49)"
  "api.database$fn__37989$fn__37990.invoke(database.clj:380)"
  "api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:227)"
  "api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:222)"
  "api.database$fn__37989.invokeStatic(database.clj:375)"
  "api.database$fn__37989.invoke(database.clj:375)"
  "middleware$enforce_authentication$fn__35045.invoke(middleware.clj:122)"
  "api.routes$fn__40168.invokeStatic(routes.clj:59)"
  "api.routes$fn__40168.invoke(routes.clj:59)"
  "routes$fn__40246$fn__40247.doInvoke(routes.clj:75)"
  "routes$fn__40246.invokeStatic(routes.clj:71)"
  "routes$fn__40246.invoke(routes.clj:71)"
  "middleware$log_api_call$fn__35144$fn__35146.invoke(middleware.clj:331)"
  "middleware$log_api_call$fn__35144.invoke(middleware.clj:330)"
  "middleware$add_security_headers$fn__35094.invoke(middleware.clj:245)"
  "middleware$bind_current_user$fn__35049.invoke(middleware.clj:142)"
  "middleware$maybe_set_site_url$fn__35098.invoke(middleware.clj:268)"],
 :sql-exception-chain
 ["MySQLIntegrityConstraintViolationException:"
  "Message: Cannot delete or update a parent row: a foreign key constraint fails (`metabase`.`metabase_field`, CONSTRAINT `fk_field_ref_table_id` FOREIGN KEY (`table_id`) REFERENCES `metabase_table` (`id`))"
  "SQLState: 23000"
  "Error Code: 1451"]}

08-07 11:56:27 ERROR metabase.middleware :: DELETE /api/database/6 500 (899 ms) (55 DB calls)
{:message
 "Cannot delete or update a parent row: a foreign key constraint fails (`metabase`.`metabase_field`, CONSTRAINT `fk_field_ref_table_id` FOREIGN KEY (`table_id`) REFERENCES `metabase_table` (`id`))",
 :stacktrace
 ["models.database$pre_delete.invokeStatic(database.clj:52)"
  "models.database$pre_delete.invoke(database.clj:49)"
  "api.database$fn__37989$fn__37990.invoke(database.clj:380)"
  "api.common.internal$do_with_caught_api_exceptions.invokeStatic(internal.clj:227)"
  "api.common.internal$do_with_caught_api_exceptions.invoke(internal.clj:222)"
  "api.database$fn__37989.invokeStatic(database.clj:375)"
  "api.database$fn__37989.invoke(database.clj:375)"
  "middleware$enforce_authentication$fn__35045.invoke(middleware.clj:122)"
  "api.routes$fn__40168.invokeStatic(routes.clj:59)"
  "api.routes$fn__40168.invoke(routes.clj:59)"
  "routes$fn__40246$fn__40247.doInvoke(routes.clj:75)"
  "routes$fn__40246.invokeStatic(routes.clj:71)"
  "routes$fn__40246.invoke(routes.clj:71)"
  "middleware$log_api_call$fn__35144$fn__35146.invoke(middleware.clj:331)"
  "middleware$log_api_call$fn__35144.invoke(middleware.clj:330)"
  "middleware$add_security_headers$fn__35094.invoke(middleware.clj:245)"
  "middleware$bind_current_user$fn__35049.invoke(middleware.clj:142)"
  "middleware$maybe_set_site_url$fn__35098.invoke(middleware.clj:268)"],
 :sql-exception-chain
 ["MySQLIntegrityConstraintViolationException:"
  "Message: Cannot delete or update a parent row: a foreign key constraint fails (`metabase`.`metabase_field`, CONSTRAINT `fk_field_ref_table_id` FOREIGN KEY (`table_id`) REFERENCES `metabase_table` (`id`))"
  "SQLState: 23000"
  "Error Code: 1451"]}
```

Metabase is using MySQL v5.7

**Comment by senior** [comment]
I'm having difficulty reproducing this issue. Have you tried deleting it again? The only thing @camsaul and I thought of that could cause this was maybe a sync of the redshift database going the same time the database was being deleted in Metabase. If you can try it again and it works, that's probably the issue.

**Comment by Slayr** [comment]
I have been using metabase for over an year and this is similar to the issue I face as well, large databases do not get deleted - we have a MongoDB database with around 300 collections and a MySQL database with ~ 100 tables(40GB data). 

In the older version, there was no message and the page used to hang up but now this is the only error I get - "Server error encountered"

**Comment by davivc** [comment]
I have a similar issue as well. We have a Postgres Database with almost 2000 schemas but when I try to delete it metabase always returns "Server error encountered". The thing is, that the user has changed (and we've stopped to use multiple schemas too) and I don´t have access to this database anymore, so we just need to delete it from our metabase.

**Comment by flamber** [comment]
This was likely fixed in 0.32.0 by #9626. Please comment if you're still seeing this - and include version info and logs.
